lib/pliny/error_reporter.rb

```diff
@@ -0,0 +1,42 @@
+require 'rollbar/exception_reporter'
+require 'rollbar/request_data_extractor'
+
+class Pliny::ErrorReporter
+  def self.notify(exception, context: {}, rack_env: {})
+    Pliny.log_exception(exception)
+
+    REPORTERS.each do |reporter|
+      begin
+        reporter.new.notify(exception, context: context, rack_env: rack_env)
+      rescue
+        Pliny.log_exception($!)
+      end
+    end
+  end
+
+  class RollbarReporter
+    include ::Rollbar::ExceptionReporter
+    include ::Rollbar::RequestDataExtractor
+
+    def notify(exception, context:, rack_env:)
+      Rollbar.reset_notifier!
+      scope = fetch_scope(context: context, rack_env: rack_env)
+      Rollbar.scoped(scope) do
+        report_exception_to_rollbar(rack_env, exception)
+      end
+    end
+
+    private
+
+    def fetch_scope(context:, rack_env:)
+      {
+        request: proc { extract_request_data_from_rack(rack_env) }
+      }
+    rescue Exception => e
+      report_exception_to_rollbar(rack_env, e)
+      raise
+    end
+  end
+
+  REPORTERS = [RollbarReporter]
```

Comment:
It might make sense to make this configurable at some point in the future, but not too big of a deal as long as there's still one reporting option.

It's possible too that we'll establish a convention for this with the metrics reporter.